File: fashionist/routes/wardrobe.py
from fastapi import APIRouter, Response
from ..models.wardrobe import Wardrobe, UpdateWardrobe

router = APIRouter(prefix='/wardrobe/{_id}')


@router.get('/', status_code=200, response_model=Wardrobe)
async def get_wardrobe(
        _id: str,
        outfits_offset: int = 0,
        outfits_limit: int = 10,
        clothes_offset: int = 0,
        clothes_limit: int = 10
):
    return (await Wardrobe.read_extra(
        query={"_id": _id},
        extra={
            "outfits": {
                "$slice": [outfits_offset, outfits_limit]
            },
            "clothes": {
                "$slice": [clothes_offset, clothes_limit]
            }
        }))[0]


@router.post('/', status_code=201, response_model=Wardrobe)
async def create_wardrobe(wardrobe: Wardrobe):
    await Wardrobe.create(wardrobe)
    return wardrobe


@router.put('/{_id}', status_code=204, response_class=Response)
async def update_wardrobe(
        _id: str,
        wardrobe: UpdateWardrobe
):
    await Wardrobe.update(_id, wardrobe)


@router.delete('/{_id}', status_code=204, response_class=Response)
async def delete_wardrobe(_id: str):
    await Wardrobe.delete(_id)

File: fashionist-front/src/services/Wardrobe.js
import client from './Base';

const endpoint = '/wardrobes';


export async function getWardrobe(id, outfitsOffset, outfitsLimit, clothesOffset, clothesLimit) {
    let url = `${endpoint}/${id}`;
    url = urlParamCheck(url, 'outfits_offset', outfitsOffset);
    url = urlParamCheck(url, 'outfits_limit', outfitsLimit);
    url = urlParamCheck(url, 'clothes_offset', clothesOffset);
    url = urlParamCheck(url, 'clothes_limit', clothesLimit);

    return new Promise((resolve, reject) => {
        client.get(url)
            .then((response) => {
                resolve(response.data);
            })
            .catch((error) => {
                reject(new Error(error));
            });
    });
}

export async function createWardrobe(clothes, outfits) {
    return new Promise((resolve, reject) => {
        client.post(endpoint, { clothes, outfits })
            .then((response) => {
                resolve(response.data);
            })
            .catch((error) => {
                reject(new Error(error));
            });
    });
}

export async function updateWardrobe(id, clothes, outfits) {
    return new Promise((resolve, reject) => {
        client.put(`endpoint/${id}`, { clothes, outfits })
            .then((response) => {
                resolve(response.data);
            })
            .catch((error) => {
                reject(new Error(error));
            });
    });
}

export async function deleteWardrobe(id) {
    return new Promise((resolve, reject) => {
        client.delete(`endpoint/${id}`)
            .then((response) => {
                resolve(response.data);
            })
            .catch((error) => {
                reject(new Error(error));
            });
    });
}